### ISSUE TITLE
Fast-Downward SSL reporting 404

### ISSUE THREAD

**Issue by pet1330** [LCAS/teaching#28]
@marc-hanheide: The `https` version of this site reports a 404 error. Would it be possible to add the SSL certificate to this site to allow the `https` version to be used (it would be even better to force a user redirect to this version).

**Comment by marc-hanheide** [comment]
tried to do exactly that: https://lcas.lincoln.ac.uk/fast-downward/

but the problem is that somehow the assets link include a hard-coded protocol...

```html
  <link href="http://lcas.lincoln.ac.uk/fast-downward/static/css/8129185d87d72d8390f6.css" rel="stylesheet">
  <script src="http://lcas.lincoln.ac.uk/fast-downward/static/js/8129185d87d72d8390f6.js"></script>
```

Any idea, why?

**Comment by pet1330** [comment]
It appears as though the `url_for` command that we are using, should respond with the same protocol as the request, but if it cannot identify the protocol type then it defaults to apps `PREFERRED_URL_SCHEME` config value which by default is `http`.

Because the app is behind the reverse proxy, the request headers might be different from the one the end user is sending, making them appear as `http` headers.

Try changing to see if it works:
```diff
if __name__ == '__main__':
    app.register_blueprint(bp, url_prefix='/fast-downward')
+     app.config.update(dict(PREFERRED_URL_SCHEME='https'))
    app.run(debug=True, threaded=True, host='0.0.0.0')
```

**Comment by pet1330** [comment]
Alternatively, try changing line 50 from:
```
return request.url_root + config[asset]
```

to be 

```
return 'https://' + request.host + request.path + config[asset]
```


https://github.com/LCAS/teaching/blob/c1b77026d095e09bbf295a3ebc7766387b2bf214/fd_webserver/app.py#L44-L51

**Comment by marc-hanheide** [comment]
closed in 73d29fa197b7c88cd64963deef2e121b4ed13d87
